From abc7ca2b381b2655d42a680dc754952aadc331e7 Mon Sep 17 00:00:00 2001
From: "/setup.sh" <support@vision-components.com>
Date: Fri, 26 Nov 2021 13:23:26 +0100
Subject: [PATCH] Bugfix in destroy_buffer_table(). When the
 capture_buffer_table isn't created before, the function crashed because of a
 NULL pointer dereferencing.

---
 .../drivers/media/platform/tegra/camera/capture_common.c       | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kernel/nvidia/drivers/media/platform/tegra/camera/capture_common.c b/kernel/nvidia/drivers/media/platform/tegra/camera/capture_common.c
index 9a9525f..9c4c1b6 100644
--- a/kernel/nvidia/drivers/media/platform/tegra/camera/capture_common.c
+++ b/kernel/nvidia/drivers/media/platform/tegra/camera/capture_common.c
@@ -96,6 +96,9 @@ void destroy_buffer_table(struct capture_buffer_table *tab)
 	struct hlist_node *next;
 	struct capture_mapping *pin;
 
+	if (!tab) 
+		return;
+
 	write_lock(&tab->hlock);
 
 	hash_for_each_safe(tab->hhead, bkt, next, pin, hnode) {
-- 
2.25.1

